bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -239,28 +239,35 @@ public String authority() {
    * @param factory a {@link com.google.cloud.bigtable.grpc.io.ChannelPool.ChannelFactory} object.
    * @throws java.io.IOException if any.
    */
-  public ChannelPool(List<HeaderInterceptor> headerInterceptors, ChannelFactory factory)
+  public ChannelPool(List<HeaderInterceptor> headerInterceptors, ManagedChannel channel)
       throws IOException {
-    this(headerInterceptors, factory, 1);
+    this(headerInterceptors, Collections.singletonList(channel));
   }
 
-
   /**
    * <p>Constructor for ChannelPool.</p>
    *
    * @param headerInterceptors a {@link java.util.List} object.
-   * @param factory a {@link com.google.cloud.bigtable.grpc.io.ChannelPool.ChannelFactory} object.
+   * @param channels a {@link List} of {@link ManagedChannel}s.
    * @throws java.io.IOException if any.
    */
-  public ChannelPool(List<HeaderInterceptor> headerInterceptors, ChannelFactory factory, int count)
+  public ChannelPool(List<HeaderInterceptor> headerInterceptors, List<ManagedChannel> channels)
       throws IOException {
-    Preconditions.checkArgument(count > 0, "Channel count has to be a positive number.");
-    ImmutableList.Builder<InstrumentedChannel> channeListBuilder = ImmutableList.builder();
-    for (int i = 0; i < count; i++) {
-      channeListBuilder.add(new InstrumentedChannel(factory.create()));
-    }
-    this.channels = channeListBuilder.build();
-    authority = channels.get(0).authority();
+    this(headerInterceptors, toInstrummentedChannels(channels));
+  }
+
+  /**
+   * <p>
+   * Constructor for ChannelPool.
+   * </p>
+   * @param headerInterceptors a {@link java.util.List} object.
+   * @param channels a {@link ImmutableList} of {@link InstrumentedChannel}s.
+   * @throws java.io.IOException if any.
+   */
+  public ChannelPool(List<HeaderInterceptor> headerInterceptors,
+      ImmutableList<InstrumentedChannel> channels) throws IOException {
```

Comment:
Similar to above: might be better to keep this as a private implementation detail